internal/pkg/manifest/job.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package manifest provides functionality to create Manifest files.
+package manifest
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+)
+
+const (
+	// ScheduledJobType is a recurring ECS Fargate task which runs on a schedule.
+	ScheduledJobType = "Scheduled Job"
+)
+
+const (
+	scheduledJobManifestPath = "workloads/jobs/scheduled-job/manifest.yml"
+)
+
+// JobTypes holds the valid job "architectures"
+var JobTypes = []string{
+	ScheduledJobType,
+}
+
+// ScheduledJob holds the configuration to build a container image that is run
+// periodically in a given environment with timeout and retry logic.
+type ScheduledJob struct {
+	Service            `yaml:",inline"`
+	ScheduledJobConfig `yaml:",inline"`
+	Environments       map[string]*ScheduledJobConfig `yaml:",flow"`
+
+	parser template.Parser
+}
+
+// ScheduledJobConfig holds the configuration for a scheduled job
+type ScheduledJobConfig struct {
+	Image          ServiceImage `yaml:",flow"`
+	TaskConfig     `yaml:",inline"`
+	*LogConfig     `yaml:"logging,flow"`
+	Sidecar        `yaml:",inline"`
+	ScheduleConfig `yaml:",inline"`
+}
+
+// ScheduleConfig holds the fields necessary to describe a scheduled job's execution frequency and error handling.
+type ScheduleConfig struct {
+	Schedule string `yaml:"schedule"`
+	Timeout  string `yaml:"timeout"`
+	Retries  int    `yaml:"retries"`
```

Comment:
Is `Retries=0` considered as a valid value? If so then it might not be written to the manifest as expected 🤔

Reply:
maybe we can just ignore it if they set `Retries` to 0.

Reply:
I think it's probably best to swallow it--since the cf template will only add retry logic if retries is nonzero, it will make no difference in the CF if in the manifest `retries` is 0 or nonexistent.